From c1f29cabc8a433fcf2285a1b9bf3a882b28e77de Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Sun, 21 Nov 2021 10:37:25 +0000
Subject: [PATCH] GCC: fix name clash of blink::ContainerQuery

blink::ContainerQuery is a class and at the same time a method
in blink::CSSContainerRule which returns the class. GCC errors
out, because it cannot distinguish between both. Add an explicit
class to the return type to avoid confusion.
---
 third_party/blink/renderer/core/css/css_container_rule.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third_party/blink/renderer/core/css/css_container_rule.h b/third_party/blink/renderer/core/css/css_container_rule.h
index 55dd56c..401504c 100644
--- a/third_party/blink/renderer/core/css/css_container_rule.h
+++ b/third_party/blink/renderer/core/css/css_container_rule.h
@@ -27,7 +27,7 @@ class CSSContainerRule final : public CSSConditionRule {
 
  private:
   CSSRule::Type GetType() const override { return kContainerRule; }
-  const ContainerQuery& ContainerQuery() const;
+  const class ContainerQuery& ContainerQuery() const;
 };
 
 template <>
-- 
2.32.0

